### ISSUE TITLE
Ensure kyma system components have sidecar injected

### ISSUE THREAD

**Issue by piotrmsc** [kyma-project/kyma#2072]
**Description**
(Update)
With https://github.com/kyma-project/kyma/issues/2073 we will introduce default sidecar injection in the kyma. We have to ensure that kyma-system components that do not already have a sidecar will remain without it, components with sidecar must still have it.

**Reasons**

Sidecars are required to make use of service-mesh functionality.

All kyma components which do not have sidecars should remain unchanged after enabling the policy - `sidecar.istio.io/inject: "false"` annotation should be added to such deployments. 

**Attachments**

Acceptance Criteria:

- Kyma components without envoy remain without it
- Kyma examples updated if necessary

**Comment by sjanota** [comment]
I suggest other approach.
1. All Kyma components have explicit  `sidecar.istio.io/inject` annotation either true or false. Thanks to that Kyma operations no longer depends on injection policy.
2. Because policy is no longer a requirement for Kyma to run we can move it from patch to overrides. Patch is greatly simplified thanks to that.
3. When someone decides to install Kyma using their own Istio they have the freedom to choose injection policy. Policy is set to `enabled` by default, so one need to intentionally disable it.

**Comment by piotrmsc** [comment]
Thx for your thoughts ;) 
In the istio patch we have to ensure the policy is enabled, policy enablement, however, can be done via override `{{ .Values.global.proxy.autoInject }}`. All kyma-system components that do not have sidecar should have annotation added. Injection should be enabled by default because we heavily rely on istio functionality. Users get `batteries included` with kyma, and have monitoring, tracing security and other feature enabled by default with the possibility to disable them if they want to.

**Comment by Tomasz-Smelcerz-SAP** [comment]
One safe way to proceed is:

Step 1:
- Identify objects that can possibly create a Pod (istio injection target): Deployments, StatefulSets, ReplicaSets, Pods, etc.
- Add ```sidecar.istio.io/inject: "false"``` to every object that does not have it yet. For now, the policy is disabled, so this step should not change anything.
- To improve the review process, make separate PR's for every Kyma component. Lot's of PR's, but will be small, easy to review by owners.

After Step 1 is done, every component is explicitly labeled with ```sidecar.istio.io/inject``` annotation and Kyma is working as before.

Step 2:
- Separate PR that switches default istio policy from disabled to enabled.

Because every component is already labeled explicitly, this step should not actually change anything for existing components. But new Deployments will have sidecars injected automatically as required.

Step 3:
- Documentation changes.

**Comment by Tomasz-Smelcerz-SAP** [comment]
About knative installation:
Knative is installed in three namespaces: `knative-build`, `knative-serving` and `knative-eventing`. Only `knative-serving` namespace is labeled for sidecar injection, so it's enough to ensure all involved resources from this namespace have explicit `sidecar.istio.io/inject` annotation.

**Comment by piotrmsc** [comment]
@kfurgol  @Tomasz-Smelcerz-SAP  The sidecar is injected to dex pod although previously it was not. Take a look[here](https://github.com/kyma-project/kyma/blob/master/resources/dex/templates/dex-deployment.yaml#L21). 

If dex is working correctly with a sidecar on a minikube and on a cluster then proper cleanup should be done.
